### ISSUE TITLE
It should not throw 404 error with message of "not found". >>> (Regarding  "Updating data returns an empty Exception when there is no results returned #87") 

### ISSUE THREAD

**Issue by JasonChiu-dev** [supabase/supabase-flutter#315]
The library now throws 404 error with message of `not found` now, which is the expected behavior of Postgrest when there are no rows to update, so closing this PR as it's resolved!

_Originally posted by @dshukertjr in https://github.com/supabase/supabase-flutter/issues/87#issuecomment-1263261813_
      
It is so strange to throws 404 error with message of not found!!
On SQL command test, if the data is not found when doing "update" action, the SQL response "Success. No rows returned". It should not throws 404 error with message of not found.

**Comment by JasonChiu-dev** [comment]
For example, when I have the major table for the basic data (e.g. customers.customerId and customers.name) and then I have the relative table (orders) for the detail data (e.g. orders.numer, orders.customerId (foreign key relate to customer.customerId) ...) ...etc.

It is not necessary that I have to check  table (orders) every time when I found I mis-typed the customers.customerId data and try to update customers table and orders table (maybe there were many records in orders table with the mis-typed the customers.customerId, or none  record in orders table with the mis-typed the customers.customerId. But I just want to update the orders table when I update the customers table with the correct customers.customerId).

For your information.

Jason

**Comment by JasonChiu-dev** [comment]
@dshukertjr 
Is there anyone who can help?

**Comment by dshukertjr** [comment]
@JasonChiu-dev you are saying that it should just return an empty array without throwing an error, correct?

supabase-flutter is just reflecting what Postgrest returns us, so you might want to open an issue there for this.

**Comment by JasonChiu-dev** [comment]
Hi Tyler,

Yes, it should just return an empty array without throwing an error. My
apps (with Supabase support) already running for a period of time, the
Supabase react very well.

My customer recently reported that the app was not working properly, and I
checked and found that this is the problem as I reported.

In the past, I used supabase-dart and work correctly. Recently, I changed
to supabase-flutter. Could it be that Supabase has been modified after the
updated version to cause this problem?

Finally,  it should just return an empty array without throwing an error.

Best,
Jason



Tyler ***@***.***> 於 2022年12月23日 週五 16:53 寫道：

**Comment by dshukertjr** [comment]
@JasonChiu-dev 
Actually, I was wrong, and the js client library does have a mechanism to ignore 404 errors. 
https://github.com/supabase/postgrest-js/issues/295

Will fix this and ship an update ASAP!

**Comment by JasonChiu-dev** [comment]
Great, thanks!

Best,
Jason

Tyler ***@***.***>於 2022年12月23日 週五，下午8:34寫道：

**Comment by dshukertjr** [comment]
@JasonChiu-dev 
Patch for this issue should be deployed on all Supabase projects, but could you check if it is still happening?
https://github.com/PostgREST/postgrest/issues/2343

**Comment by JasonChiu-dev** [comment]
> 



> @JasonChiu-dev Patch for this issue should be deployed on all Supabase projects, but could you check if it is still happening? [PostgREST/postgrest#2343](https://github.com/PostgREST/postgrest/issues/2343)
The same situation still exists. I still get the same error message.

<img width="239" alt="截圖 2022-12-26 下午12 09 34" src="https://user-images.githubusercontent.com/81674344/209498985-e36142f3-8c9b-4e70-acc1-ce43d13b8d3c.png">

**Comment by dshukertjr** [comment]
@JasonChiu-dev 
Can you contact support@supabase.com about this with your project reference?

**Comment by steve-chavez** [comment]
@JasonChiu-dev You can check your postgrest version as described [here](https://github.com/supabase/postgrest-js/issues/223#issuecomment-1333137399).

If you have a version >= [v10.1.0](https://github.com/PostgREST/postgrest/releases/tag/v10.1.0), the 404 error on update should not happen.

**Comment by JasonChiu-dev** [comment]
> @JasonChiu-dev You can check your postgrest version as described [here](https://github.com/supabase/postgrest-js/issues/223#issuecomment-1333137399).
> 
> If you have a version >= [v10.1.0](https://github.com/PostgREST/postgrest/releases/tag/v10.1.0), the 404 error on update should not happen.

This is correct. When I followed the supabase support team's guidance: "**if you pause & restore your project, this will upgrade the PostgREST version, many of the Postgres extensions and Postgres to version 15.
https://supabase.com/docs/guides/platform/migrating-and-upgrading-projects#upgrade-your-project**", it is really working.

[https://supabase.com/docs/guides/platform/migrating-and-upgrading-projects#upgrade-your-project](url)

Now, my project upgrade to the PostgREST version 10.1.0 and the 404 error on update do not happen.

I appreciated.
